tools/processresults.py

```diff
@@ -0,0 +1,307 @@
+"""Functions for processing CUES data in AppEngine JSON format into CSV format
+
+Has one public function, ProcessResults, used like this from interpreter:
+    ProcessResults('2015-01-06cuesDogfoodData.json', '2015-01-06-')
+"""
+
+import csv
+import dateutil.parser
+from datetime import datetime
+import json
+import re
+
+DOGFOOD_START_DATE = datetime(2014, 12, 01, 0, 0, 0, 0)
+CONDITIONS = [
+    'ssl-overridable-proceed', 'ssl-overridable-noproceed',
+    'ssl-nonoverridable', 'malware-proceed', 'malware-noproceed',
+    'phishing-proceed', 'phishing-noproceed', 'extension-proceed',
+    'extension-noproceed']
+ATTRIBUTE_QUESTION_PREFIX = 'To what degree do each of the following'
+
+def ProcessResults(json_in_file, csv_prefix):
+  """Take results from AppEngine JSON file, process, and write to CSV file.
+
+  Results from the input JSON file will be filtered into the 9 experimental
+  conditions and written to CSV output files named in the format
+  <csv_prefix><condition>.csv. CSV column headers will be the meta data fields
+  date_received, date_taken, participant_id, and survey_type, plus the
+  questions asked of the participant (with [URL] replacing actual URLs in the
+  question text). Responses where questions were replaced with PLACEHOLDER
+  will be ignored and not copied to the CSV file.
+
+  Args:
+    json_in_file: File with raw AppEngine results in JSON format
+    csv_prefix: Prefix to output CSV files. These files will be named
+        in the format <csv_prefix><condition>.csv. For example, if
+        csv_prefix is '2015-01-15-' then one output file would be
+        named 2015-01-15-ssl-overridable-proceed.csv
+
+  Returns:
+    None
+  """
+  parsed_events = _ParseSurveyResults(json_in_file)
+  parsed_events = _DiscardResultsBeforeDate(parsed_events, DOGFOOD_START_DATE)
+
+  for c in CONDITIONS:
+    try:
+      filtered_results = _FilterByCondition(c, parsed_events)
+      canonical_index = _CanonicalizeQuestions(filtered_results)
+      _WriteToCsv(filtered_results, canonical_index, csv_prefix + c + '.csv')
+    # Print UnexpectedFormatException and continue, since they are usually
```

Comment:
nit: I feel like this comment should go below except statement with an indentation.